_posts/2019-06-03-deploy-docker-to-gcp.asciidoc

```diff
@@ -0,0 +1,21 @@
+---
+layout: post
+title: Deploying a Dockerized App on GCP
+logoUrl: /assets/img/guides/docker_image.png
+excerpt: Before we can deploy a dockerized app, we need to first create one. For the purposes of this guide we will create a basic Node.js app that responds to requests on port 8080.
+---
```

Comment:
AsciiDoc supports its own syntax for front matter. See the VPC post for an example. Does that work with the Jekyll AsciiDoc plugin?

Reply:
It does but doesn't get rendered if `page` not prefixed before every variable e.g :page-title :page-image. That was why I felt we should just use the font matter.

Reply:
We don't need the front matter to be rendered though... We just need it for the metadata. The rendering of that metadata would be done by the layout, right?

Or are you saying that you can't access the metadata in the layout unless that metadata is prefixed by `:page`?

Reply:
The latter. We cannot access the metadata if not prefixed by `:page`

Reply:
Ah, OK... Any reason not to prefix it then? Seems like an easy enough standard to follow?

Reply:
Yeah. Done that but there are still a few issues with excerpt,  image, and categories. Having similar issues and stated in these links https://github.com/asciidoctor/jekyll-asciidoc/issues/200 https://github.com/asciidoctor/jekyll-asciidoc/issues/149 and the solutions proposed here still doesn't fix the issue. So for those exceptions, I use front matter

Reply:
Understood!